### ISSUE TITLE
Dimension mismatch training with my own model / why does training give the same prediction for all inputs?

### ISSUE THREAD

**Issue by johnswan** [BVLC/caffe#261]
Hi,

I gave up on wrapper.py as there seemed to be a bug where it was sending the wrong type to Pycaffe (which attempts to call .size on a list and gets 0). 

However, a short (the simplest possible?) prediction script causes the same error:

F0326 09:37:00.348891 2102817152 pycaffe.cpp:127] Check failed: len(bottom) == input_blobs.size() (1 vs. 0) 

This is the script:

``` python
from caffe import imagenet
# from matplotlib import pyplot

# Set the right path to your model file, pretrained model,
# and the image you would like to classify.
MODEL_FILE = 'train.prototxt'
PRETRAINED = 'train_iter_10000'
IMAGE_FILE = 'image.jpg'

net = imagenet.ImageNetClassifier(
    MODEL_FILE, PRETRAINED)

net.caffenet.set_phase_test()
# net.caffenet.set_mode_cpu()
net.caffenet.set_mode_gpu()

prediction = net.predict(IMAGE_FILE)
print '\nprediction:'
print prediction
print '\n\nprediction shape:', prediction.shape
pyplot.plot(prediction)
```

Is it me?

Any help gratefully received.

Best regards
John

**Comment by wkiri** [comment]
I had the same problem, and I tracked it down to a different cause.

I had renamed the final layer in my train_val.prototxt network from "fc8" to "fc8_landmarks" (specific to my image data set.  However, I had forgotten to do the same thing in deploy.txt (it was still called "fc8").  I didn't get any error messages, but all of my classes had the same predicted probability for all items.  By changing the layer name to "fc8_landmarks" in deploy.txt (to match train_val.prototxt), I was able to get sensible output.

**Comment by nrajani** [comment]
Hi 
I am facing a related issue and would appreciate any help in this. 

I modified the ResNet prototxt file to have the ROI proposals and was able to fine tune it successfully in 50K iterations on val set 
`Train net output #3: rpn_loss_bbox = 0.0179207 (* 1 = 0.0179207 loss)`
Now I am trying to use the final caffe model along with the deploy.prototxt for ResNet modified as a test prototxt (removing lr params, top layer and appending the input data layer). However I keep getting this error.
`File "./lib/rpn/anchor_target_layer.py", line 116, in forward (all_anchors[:, 2] < im_info[0][1] ) & # width ValueError: operands could not be broadcast together with shapes (17100,) (600,800)`

I printed a few debug statements but was unsuccessful in trouble shooting. Any help would be appreciated.
